tests/unit/xgb/test_xgboost.py

```diff
@@ -22,11 +22,14 @@
 import sklearn.datasets
 import xgboost
 
-from merlin.core.dispatch import HAS_GPU
+from merlin.core.compat import HAS_GPU
 from merlin.datasets.synthetic import generate_data
 from merlin.io import Dataset
 from merlin.models.xgb import XGBoost, dataset_to_xy
 
+if not HAS_GPU:
+    pytest.skip(reason="No GPU available", allow_module_level=True)
```

Comment:
So update here, I have been able to make the change to where all non-gpu tests run regardless of environment. And only added the skipif decorator to the GPU test. Apparently the dask client requires the --privileged flag. Cant really explain why since that flag opens up pretty much everything.

Reply:
No its not new... I just wasn't using it in my dev environment so I was hitting hangs. I think its really something to do with IPC being set correctly so dask workers can talk to each other but that investigation can be left for another PR.